src/parser/core/modules/GlobalCooldown.js

```diff
@@ -59,20 +70,40 @@ export default class GlobalCooldown extends Module {
 				break
 
 			case 'cast':
-				if (this._castingEvent && this._castingEvent.ability.guid === action.id) {
-					this.saveGcd(this._castingEvent, false)
-				} else {
-					this.saveGcd(event, true)
-				}
+				const hasBeginCast = this._castingEvent !== null && this._castingEvent.ability.guid === action.id
+				const relevantEvent = hasBeginCast ? this._castingEvent : event
+				this.saveGcd(this._lastGcd, relevantEvent.timestamp) // Save last gcd with current timestamp
```

Comment:
Objects pass by reference. While the implementation of `saveGcd` _currently_ doesn't get effected by that, I can see future changes might hit that, and it'd be a pretty nasty debug to run through.

Gonna ask for the `this._lastGcd` to be passed as a copy to prevent those potential down-road issues - simplest impl. would be `{...this._lastGcd}`.

Reply:
Will make these pass as copy changes